src/meta-srv/src/service/admin/region_migration.rs

```diff
@@ -99,10 +101,41 @@ impl SubmitRegionMigrationTaskHandler {
     /// Submits a region migration task, returns the procedure id.
     async fn handle_submit(
         &self,
-        _task: SubmitRegionMigrationTaskRequest,
+        task: SubmitRegionMigrationTaskRequest,
     ) -> Result<SubmitRegionMigrationTaskResponse> {
-        // TODO(weny): waits for https://github.com/GreptimeTeam/greptimedb/pull/3014
-        todo!()
+        let SubmitRegionMigrationTaskRequest {
+            cluster_id,
+            region_id,
+            from_peer_id,
+            to_peer_id,
+        } = task;
+
+        let from_peer = self.peer_lookup.peer(cluster_id, from_peer_id).context(
```

Comment:
Do we really need a "peer lookup" functionality like this? If the user submits a region migration request explicitly like this, he must have known what he is doing, and we shall not hinder it - at least not like this, hinder it because of some fragile heartbeats say the peer is not available. Were the region migration not able to proceed because the peers are not available, simply returns the errors to the user.

On the other hand, if the region migrations are submitted by metasrv itself by some load balance method, the `to_peer` should be selected by `Selector`. This "peer lookup" is of no use there.

cc @fengjiachun

Reply:
> On the other hand, if the region migrations are submitted by metasrv itself by some load balance method, the to_peer should be selected by Selector. This "peer lookup" is of no use there.

Yes. If we introduce the `some load balance method` later, we can only check whether the `from_peer` is available.

Reply:
I got you. We should allow users to migrate regions from a failed node to another place? How about adding a `force` flag to allow users to do this?

Reply:
I think this is just to ensure that the 'from' and 'to' nodes are alive, that they can be commanded, similar to a pre-check. If the check fails, we can actually foresee that the subsequent operations will fail and we can report an error in advance. However, I do agree that Peer Lookup does not seem necessary because we already have similar functionality cc @WenyXu : https://github.com/GreptimeTeam/greptimedb/blob/840e94630d01818696538dbd2f28441702c1ccc9/src/meta-srv/src/handler/keep_lease_handler.rs#L29

Reply:
`KeepLeaseHandler` saves the time of the latest heartbeat received from a corresponding node. Based on this time, we can make some judgements, such as evaluating the activity level of this node. Compared to `Peer Lookup`, which deregisters after a single heartbeat disruption, this method is more robust.
In fact, `Selector` uses this: https://github.com/GreptimeTeam/greptimedb/blob/da68d8ce4b0e3f6f5314de5ef361268812632098/src/meta-srv/src/selector/lease_based.rs#L39